evalml/automl/automl_search.py

```diff
@@ -433,33 +417,6 @@ def _validate_problem_configuration(self, problem_configuration=None):
                                  f"parameters. Received {problem_configuration}.")
         return problem_configuration or {}
 
-    def _validate_data_checks(self, data_checks):
-        """Validate data_checks parameter.
-
-        Arguments:
-            data_checks (DataChecks, list(Datacheck), str, None): Input to validate. If not of the right type,
-                raise an exception.
-
-        Returns:
-            An instance of DataChecks used to perform checks before search.
-        """
-        if isinstance(data_checks, DataChecks):
-            return data_checks
-        elif isinstance(data_checks, list):
-            return AutoMLDataChecks(data_checks)
-        elif isinstance(data_checks, str):
-            if data_checks == "auto":
-                return DefaultDataChecks(problem_type=self.problem_type, objective=self.objective, n_splits=self.data_splitter.get_n_splits())
-            elif data_checks == "disabled":
-                return EmptyDataChecks()
-            else:
-                raise ValueError("If data_checks is a string, it must be either 'auto' or 'disabled'. "
-                                 f"Received '{data_checks}'.")
-        elif data_checks is None:
-            return EmptyDataChecks()
-        else:
-            return DataChecks(data_checks)
```

Comment:
I think its fine to delete this today. I could see us lifting/parroting this code in the future for the data check actions helper.